File: backend/src/main/kotlin/com/cookbyte/backend/service/ReviewService.kt
package com.cookbyte.backend.service

import com.cookbyte.backend.domain.Recipe
import com.cookbyte.backend.domain.Review
import com.cookbyte.backend.domain.User

interface ReviewService {
    fun findAll(): List<Review>?
    fun getReviewByRecipeId(recipeId: Long): List<Review>?
    fun getReviewsByUserId(userId: Long): List<Review>?
    fun addReview(description: String, user: User, recipe: Recipe, ratingValue: Int): Review?
    fun deleteReview(reviewId: Long)
}

File: frontend/src/app/services/review.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Review} from "../interfaces/review";
import {Observable} from "rxjs";
import {Recipe} from "../interfaces/recipe";

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  url = 'http://localhost:8080/reviews'

  reviews: Review[] | undefined

  constructor(private httpClient: HttpClient) {
  }

  getAllReviews(): Observable<Review[]> {
    return this.httpClient.get<Review[]>(this.url)
  }

  getReviewForRecipe(recipeId: number): Observable<Review[]> {
    return this.httpClient.get<Review[]>(`${this.url}/${recipeId}`)
  }

  getReviewsForUser(userId: number | undefined): Observable<Review[]> {
    return this.httpClient.get<Review[]>(`${this.url}/reviewsByUser/${userId}`);
  }
  addReview(formData: FormData): Observable<any> {
    const headers = new HttpHeaders();
    headers.set('Content-Type', 'multipart/form-data');
    return this.httpClient.post<any>(`${this.url}/addReview`, formData)
  }

  deleteReview(reviewId: number): Observable<any> {
    return this.httpClient.delete(`${this.url}/delete/${reviewId}`)
  }
}
